clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2231,9 +2263,28 @@ public void processConsumerRecord(
         && (partitionConsumptionState.getProcessedRecordSizeSinceLastSync() >= syncBytesInterval));
     defaultReadyToServeChecker.apply(partitionConsumptionState, recordsProcessedAboveSyncIntervalThreshold);
 
+    if (maybeSyncOffset) {
+      maybeSyncOffsets(record, leaderProducedRecordContext, partitionConsumptionState, subPartition);
+    }
+  }
+
+  /**
+   * Syncing offset checking in syncOffset() should be the very last step for processing a record.
+   */
+  public void maybeSyncOffsets(
+      PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> record,
+      LeaderProducedRecordContext leaderProducedRecordContext,
+      PartitionConsumptionState partitionConsumptionState,
+      int subPartition) {
+
+    long syncBytesInterval = partitionConsumptionState.isDeferredWrite()
+        ? databaseSyncBytesIntervalForDeferredWriteMode
+        : databaseSyncBytesIntervalForTransactionalMode;
+    boolean recordsProcessedAboveSyncIntervalThreshold = (syncBytesInterval > 0
+        && (partitionConsumptionState.getProcessedRecordSizeSinceLastSync() >= syncBytesInterval));
+    defaultReadyToServeChecker.apply(partitionConsumptionState, recordsProcessedAboveSyncIntervalThreshold);
```

Comment:
oh.  So the recordsProcessedAboveSyncIntervalThreshold includes the synced bytes?

Reply:
Hrm.  I guess so, that would make sense.  I'll bump it down.

Reply:
acutally.  I don't think this will get included in that calculation in any case.  So I don't think it matters?  I'll leave it though where it is because this was the ordering that was in the original code.

Reply:
Hmm, actually, it looks like if maybeSyncOffset == true here, then you are doing the sync twice?